### PR TITLE
PytatoPyOpenCLArrayContext: fix non-SVM allocator

### PR DESCRIPTION
**Pull request by matthiasdiener**

### CODE REVIEW

File: arraycontext/impl/pytato/__init__.py [hunk 1]
```diff
@@ -275,7 +275,7 @@ def __init__(
                 self.using_svm = False
 
                 from pyopencl.tools import ImmediateAllocator
-                allocator = ImmediateAllocator(queue.context)
```
[the remark this answers is not in the document]
Comment (matthiasdiener) [reply arraycontext/impl/pytato/__init__.py:278]: What do you think of https://github.com/inducer/arraycontext/pull/197/commits/b78a9ea68abae08fecb24f51ccdb9d209a48eda9 ?

File: test/test_pytato_arraycontext.py [hunk 2]
```diff
@@ -121,6 +121,72 @@ def twice(x):
...
+        cl_char.has_coarse_grain_buffer_svm = override_has_svm
+
+        actx = _PytatoPyOpenCLArrayContextForTests(base_actx.queue, allocator=alloc,
+                                               use_memory_pool=use_memory_pool)
+        f = actx.compile(twice)
+        res = actx.to_numpy(f(99))
+
+        assert res == 198
+
+        cl_char.has_coarse_grain_buffer_svm = backup_has_coarse_grain_buffer_svm
```
Comment (inducer) [inline test/test_pytato_arraycontext.py:187]: https://docs.python.org/3/library/unittest.mock.html is better for this. Suppose something crashes in this test, it'll leave the module in a weird state. (Where as `mock` handles exceptions.)

File: arraycontext/impl/pytato/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (inducer) [reply arraycontext/impl/pytato/__init__.py:278]: LGTM, just one minor nit.

**Review by inducer — APPROVED** [review]
Thanks! One minor nit, and this is ready to go.

File: test/test_pytato_arraycontext.py (hunk 2 above)
Comment (matthiasdiener) [reply test/test_pytato_arraycontext.py]: Mock is great ;-). Done in [9585727](https://github.com/inducer/arraycontext/pull/197/commits/9585727e8087ca5e14824c8aee29ba8f3e147d87)

Comment (inducer) [reply test/test_pytato_arraycontext.py]: Thanks. LGTM!

**Pull request merged**